types/fs-base-ui/html_ui/JS/Types.d.ts

```diff
@@ -0,0 +1,196 @@
+import {Feet, Latitude, Longitude} from "../../../types";
+
+declare global {
+    class LatLong {
+        constructor(latitude: Latitude, longitude: Longitude);
+        constructor(data: { lat: Latitude, long: Longitude });
+
+        __Type: "LatLong";
+        lat: Latitude;
+        long: Longitude;
+
+        /**
+         * @returns A string formatted as "40.471000, 73.580000".
+         */
+        toStringFloat(): string;
+
+        /**
+         * @returns A string formatted as "lat 40.47, long 73.58".
+         */
+        toString(): string;
+
+        /**
+         * @returns A string formatted as "4047.1N".
+         */
+        latToDegreeString(): string;
+
+        /**
+         * @returns A string formatted as "07358.0W".
+         */
+        longToDegreeString(): string;
+
+        /**
+         * @returns A string formatted as "N40°47.1 W073°58.0".
+         */
+        toDegreeString(): string;
+
+        /**
+         * @returns A string formatted as "4047.1N07358.0W".
+         */
+        toShortDegreeString(): string;
+
+        /**
+         * Parses a string into a LatLong or LatLongAlt
+         * @param str A string formatted as either "0.0,0.0" or "0.0,0.0,0.0".
+         * When the string contains spaces around the numbers, those are ignored.
+         * @returns An instance of LatLong or LatLongAlt, depending on the number of values in the string. Null when
+         * the string doesn't contain a comma.
+         */
+        static fromStringFloat(str: string): LatLong | LatLongAlt | null;
+    }
+
+    class LatLongAlt {
```

Comment:
Would it be correct to declare this as a subclass of LatLong? Then it would not be necessary to write `LatLong | LatLongAlt` for Avionics.Utils.computeDistance, Avionics.Utils.computeGreatCircleHeading, Avionics.Utils.computeGreatCircleDistance, etc.

Reply:
It is not a subtype in Asobo's code, which is why I haven't declared it as such. I did consider writing `type LatLongAltitude = LatLong | LatLongAlt` but decided against it, because I feel it is confusing to developers.